src/Domain/ElementDistribution.hpp

```diff
@@ -35,15 +47,23 @@ namespace domain {
  * (forming a zig-zag path, that under some rotation/reflection has a 'Z'
  * shape).
  *
- * Then, the first processor for a block takes as many elements have been
- * assigned to it along the first part of the Z-curve, then the second processor
- * takes its elements from the next part of the Z-curve, and so on.
- * This is a quick way of getting acceptaable spatial locality -- usually, for
- * approximately even distributions, this will ensure that elements are assigned
- * in large volume chunks, and in no more than two large clusters within the
- * block. In principle, a Hilbert curve could potentially improve upon the gains
- * obtained by this class by guaranteeing that all elements within each block
- * are orthogonally adjacent.
+ * The Morton curve method is a quick way of getting acceptable spatial locality
+ * -- usually, for approximately even distributions, it will ensure that
+ * elements are assigned in large volume chunks, and the structure of the Morton
+ * curve ensures that for a given processor and block, the elements will be
+ * assigned in no more than two orthogonally connected clusters. In principle, a
+ * Hilbert curve could potentially improve upon the gains obtained by this class
+ * by guaranteeing that all elements within each block form a single orthognally
+ * connected cluster.
+ *
+ * The assignment of portions of blocks to may take partial blocks, and/or
```

Comment:
It seems like either the `to` in `to may take` shouldn't be there or there are additional words missing